aws/rust-runtime/aws-config/src/standard_property.rs

```diff
@@ -3,23 +3,44 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+use crate::profile::{ProfileSet, PropertiesKey};
 use crate::provider_config::ProviderConfig;
 use std::borrow::Cow;
 use std::error::Error;
 use std::fmt::{Display, Formatter};
 
 #[derive(Debug)]
 pub(crate) enum PropertySource<'a> {
-    Environment { name: &'a str },
-    Profile { name: &'a str, key: &'a str },
+    Environment {
+        name: &'a str,
+        service_specific: bool,
+    },
+    Profile {
+        name: &'a str,
+        key: &'a str,
+        service_specific: bool,
+    },
 }
 
 impl Display for PropertySource<'_> {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         match self {
-            PropertySource::Environment { name } => write!(f, "environment variable `{}`", name),
-            PropertySource::Profile { name, key } => {
-                write!(f, "profile `{}`, key: `{}`", name, key)
+            PropertySource::Environment {
+                name,
+                service_specific,
+            } => {
+                #[rustfmt::skip]
+                let service_specific = if *service_specific { "service-specific " } else { "" };
```

Comment:
if you really want this on one line you could do 
`let service_specific = service_specific.then(|| "service-specific ").unwrap_or_default()`